elyra/pipeline/processor_airflow.py

```diff
@@ -0,0 +1,254 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import autopep8
+import json
+import tempfile
+import uuid
+
+from collections import OrderedDict
+
+from elyra.metadata import MetadataManager
+from elyra.pipeline import PipelineProcessor
+from elyra.util.archive import create_temp_archive
+from elyra.util.cos import CosClient
+from jinja2 import Environment, PackageLoader
+
+
+class AirflowPipelineProcessor(PipelineProcessor):
+    _type = 'airflow'
+
+    @property
+    def type(self):
+        return self._type
+
+    def process(self, pipeline):
+        unique_id = uuid.uuid4().hex[4:15]
+        if not pipeline.title:
+            pipeline_name = "pipeline" + '-' + unique_id
+        else:
+            pipeline_name = pipeline.title + '-' + unique_id
```

Comment:
For another time, but this logic should be baked into the Pipeline class via something like a `name` or 'run_name` property.